### DISCUSSION TITLE
Accomodating some time warping/distortion for motif matches

### DISCUSSION THREAD

**Discussion by ebagjones** [TDAmeritrade/stumpy#588] [Q&A] [answered]
I'm curious to know if stumpy can find near matches that have some configurable amount of temporal distortion/warping.
So for example, finding motif matches that are ~25% more compressed or dilated in time, but otherwise very close matches, or, alternatively, if a fairly complex sequence is extremely similar but a couple of the peaks are slightly closer together.

Is this something that stumpy is able to identify, or does it need to be a near perfect match in time?

**Comment by seanlaw** [comment, marked as answer]
@jefferyanderson Thank you for your question and welcome to the STUMPY community. I don't think there's an easy answer and the pragmatic thing to do is to just "try and see". Sometimes, matrix profiles may be sufficient for the job and, other times, maybe not.

What you are referring to is likely related to applying some sort of "Dynamic Time Warping" (DTW). The original matrix profile researchers are experts on this and have written a [follow up paper (that we have an open issue on)](https://github.com/TDAmeritrade/stumpy/issues/466) that aims to combine DTW and matrix profiles. Additionally, there is a [recent paper](https://www.mdpi.com/2571-9394/4/1/13/htm) that also tries to address this "variance in subsequence length" but, frankly, this is just a REALLY hard problem to solve. Needing to account for distortion/warping adds an additional (nasty) layer of computational complexity that makes it viable only for small toy datasets and nearly impractical for anything larger than, say, 10K-50K data points.

**Reply by ebagjones** [reply]
@seanlaw  thank you so much for the quick reply. I'll definitely try and see if it works for my case. I have only the most basic understanding of this domain but I can certainly appreciate how the complexity/difficulty explodes when searching across distorted/warped time.
